### PR TITLE
Fix examples (demo scripts and notebooks)

### PR DESCRIPTION
**Pull request by thomasgaudelet**
## What is the goal of this PR?
The objective of this PR is to update examples to ensure they can run.

## What are the changes implemented in this PR?
- One of the key issue was with removed support for Subset datasets in previous release of pyrelational.
- `examples/utils/` moved to `examples/demo/utils/`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 install_requires = [
     "numpy>=1.22.0",
     "pandas>=1.3.0",
-    "pytorch-lightning>=1.6",
```
Comment (paulmorio) [inline setup.py:15]: I see these changes, along with the changes in pre-commit were also made in #112 will this cause any issues with the merges?

**Review by paulmorio — APPROVED** [review]
Looks good to me, just notice that some changes are also the same in #112 will there be any merge-conflicts?

**Comment by thomasgaudelet** [comment]
> Looks good to me, just notice that some changes are also the same in #112 will there be any merge-conflicts?

@paulmorio that is because I worked from the other PR's branch to work on this. The previous PR must be merged before this one.

File: examples/demo/lightning_representative_classification.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
 import torch
 
 # Dataset and machine learning model
-from examples.utils.datasets import BreastCancerDataset  # noqa: E402
-from examples.utils.ml_models import BreastCancerClassification  # noqa: E402
+from utils.datasets import BreastCancerDataset  # noqa: E402
```
Comment (alicerdv) [inline examples/demo/lightning_representative_classification.py:13]: do we need the # noqa: E402?

Comment (paulmorio) [reply examples/demo/lightning_representative_classification.py:13]: It tells the linter to be quiet about the import layout, which I am happy with, so I'd like to say yes.

**Review by paulmorio — APPROVED** [review]
LGTM

File: examples/demo/lightning_representative_classification.py (hunk 2 above)
Comment (alicerdv) [reply examples/demo/lightning_representative_classification.py:13]: It seems strange that the linter complains specifically about this import line, but not the other ones. For this particular case, it should only be needed if there is code between the imports (https://www.flake8rules.com/rules/E402.html), but I can't see that here

Comment (thomasgaudelet) [reply examples/demo/lightning_representative_classification.py]: removed, seem to be a remnant of the past.

**Review by alicerdv — APPROVED** [review]

**Review by paulmorio — APPROVED** [review]

**Pull request merged**